From e6e30992fa4ca3e7ecf417a23a9c46af8f671226 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Salvador=20Fandi=C3=B1o?= <sfandino@yahoo.com>
Date: Fri, 27 Oct 2017 16:22:54 +0200
Subject: [PATCH] skip for C.UTF-8 too

[Simon Chopin]
In glibc 2.39, C.UTF-8 is handled the exact same way to C, including in
the gettext subsystem.

See https://sourceware.org/git/gitweb.cgi?p=glibc.git;h=d0aefec49941cf6d97e2244d6aa20bafc26d5942

Reviewed-By: Simon Chopin <simon.chopin@canonical.com>
Origin: https://github.com/vandry/Perl-Locale-gettext/pull/4
Bug: https://github.com/vandry/Perl-Locale-gettext/issues/7
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/2052930

---
 t/lib/gen_test_data.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test_data/gen_test_data.pl b/test_data/gen_test_data.pl
index 3e5fd06..ac4c21e 100644
--- a/test_data/gen_test_data.pl
+++ b/test_data/gen_test_data.pl
@@ -28,11 +28,11 @@ sub gen {
 		skip("cannot run test without a locale set", 0);
 		exit 0;
 	}
-	if ($messages eq 'C') {
+	if ($messages =~ /^C(\..*)?$/) {
 		skip("cannot run test in the C locale", 0);
 		exit 0;
 	}
-	if ($messages eq 'POSIX') {
+	if ($messages =~ /^POSIX(\..*)?$/) {
 		skip("cannot run test in the POSIX locale", 0);
 		exit 0;
 	}
